Build fails on linux due to libusb

The current version (77bdd09) fails to build on linux due to libusb.h missing
I'm on arch linux, kernel 5.7.4-8-tkg-pds and gcc 10.1.0-2 (I think the new building system still uses gcc, right?)

Anyways, here's the relevant output from the compiler:


`FAILED: src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o 
/usr/bin/c++  -DARCHITECTURE_x86_64=1 -DBOOST_ALL_NO_LIB -DBOOST_CONTEXT_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_NO_LIB -DFMT_LOCALE -DFMT_SHARED -DHAVE_SDL2 -DNDEBUG -D_FILE_OFFSET_BITS=64 -I../src/. -I../externals/./microprofile -isystem /usr/include/SDL2 -O3 -DNDEBUG   -Wall -Werror=implicit-fallthrough -Werror=missing-declarations -Werror=reorder -Wextra -Wmissing-declarations -Wno-attributes -Wno-unused-parameter -mcx16 -std=gnu++17 -MD -MT src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o -MF src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o.d -o src/input_common/CMakeFiles/input_common.dir/gcadapter/gc_adapter.cpp.o -c ../src/input_common/gcadapter/gc_adapter.cpp
In file included from ../src/input_common/gcadapter/gc_adapter.cpp:8:
../src/./input_common/gcadapter/gc_adapter.h:11:10: fatal error: libusb.h: No such file or directory
   11 | #include <libusb.h>
      |          ^~~~~~~~~~
compilation terminated.`
This appears to be related to the recent gamecube controller support.
I don't know if any more information is required, so feel free to ask for any additional info

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -333,6 +333,7 @@ endif()
 find_package(LibUSB)
 if (NOT LIBUSB_FOUND)
     add_subdirectory(externals/libusb)
+    set(LIBUSB_INCLUDE_DIR "")
     set(LIBUSB_LIBRARIES usb)
 endif()
 

--- a/src/input_common/CMakeLists.txt
+++ b/src/input_common/CMakeLists.txt
@@ -30,6 +30,7 @@ if(SDL2_FOUND)
     target_compile_definitions(input_common PRIVATE HAVE_SDL2)
 endif()
 
+target_include_directories(input_common SYSTEM PUBLIC ${LIBUSB_INCLUDE_DIR})
 target_link_libraries(input_common PUBLIC ${LIBUSB_LIBRARIES})
 
 create_target_directory_groups(input_common)

--- a/src/input_common/CMakeLists.txt
+++ b/src/input_common/CMakeLists.txt
@@ -30,8 +30,8 @@ if(SDL2_FOUND)
     target_compile_definitions(input_common PRIVATE HAVE_SDL2)
 endif()
 
-target_include_directories(input_common SYSTEM PUBLIC ${LIBUSB_INCLUDE_DIR})
-target_link_libraries(input_common PUBLIC ${LIBUSB_LIBRARIES})
+target_include_directories(input_common SYSTEM PRIVATE ${LIBUSB_INCLUDE_DIR})
+target_link_libraries(input_common PRIVATE ${LIBUSB_LIBRARIES})
 
 create_target_directory_groups(input_common)
 target_link_libraries(input_common PUBLIC core PRIVATE common Boost::boost)

--- a/src/input_common/gcadapter/gc_adapter.cpp
+++ b/src/input_common/gcadapter/gc_adapter.cpp
@@ -4,6 +4,7 @@
 
 #include <chrono>
 #include <thread>
+#include <libusb.h>
 #include "common/logging/log.h"
 #include "input_common/gcadapter/gc_adapter.h"
 

--- a/src/input_common/gcadapter/gc_adapter.h
+++ b/src/input_common/gcadapter/gc_adapter.h
@@ -8,9 +8,12 @@
 #include <mutex>
 #include <thread>
 #include <unordered_map>
-#include <libusb.h>
 #include "common/common_types.h"
 #include "common/threadsafe_queue.h"
+
+struct libusb_context;
+struct libusb_device;
+struct libusb_device_handle;
 
 namespace GCAdapter {
 

--- a/src/input_common/main.cpp
+++ b/src/input_common/main.cpp
@@ -4,7 +4,6 @@
 
 #include <memory>
 #include <thread>
-#include <libusb.h>
 #include "common/param_package.h"
 #include "input_common/analog_from_button.h"
 #include "input_common/gcadapter/gc_adapter.h"

